### PR TITLE
Add warnings to pennylane-sf plugins and expected output

### PR DESCRIPTION
**Pull request by albi3ro**
Since `pennylane-sf` is now pinned to `pennlane<=0.29`, the demos that use `pennylane-sf` are no longer automatically run.

This PR adds warnings to the top of those demos and adds the output for the demos that have recently been made non-executable.

### CODE REVIEW

File: demonstrations/quantum_neural_net.py
Comment (timmysilv) [inline demonstrations/quantum_neural_net.py:1]: this demo still needs a warning

**Review by timmysilv — APPROVED** [review]

File: demonstrations/gbs.py [hunk 1]
```diff
@@ -19,6 +19,9 @@
 
 *Authors: Josh Izaac and Nathan Killoran — Posted: 04 December 2020. Last updated: 04 December 2020.*
 
+.. warning::
+    The ``pennylane-sf`` plugin is pinned to ``pennylane<=v0.29``.
```
Comment (trbromley) [inline demonstrations/gbs.py:23]:
```suggestion
    This demo is only compatible with PennyLane version `0.29` or below.
```

File: demonstrations/gbs.py [hunk 2]
```diff
@@ -147,6 +150,22 @@
...
+#       [[ 0.23648826-0.48221431j  0.06829648+0.04447898j  0.51150074-0.09529866j
+#         0.55205719-0.35974699j]
+#        [-0.11148167+0.69780321j -0.24943828+0.08410701j  0.46705929-0.43192981j
+#          0.16220654-0.01817602j]
+#       [-0.22351926-0.25918352j  0.24364996-0.05375623j -0.09259829-0.53810588j
+#          0.27267708+0.66941977j]
+#         [ 0.11519953-0.28596729j -0.90164923-0.22099186j -0.09627758-0.13105595j
+#         -0.0200152 +0.12766128j]]
+#
 ######################################################################
```
Comment (trbromley) [inline demonstrations/gbs.py:169]: The formatting has gone odd here:
image

Do you think we need to do the following:
```suggestion
```

File: demonstrations/plugins_hybrid.py [hunk 3]
```diff
@@ -233,7 +256,35 @@ def cost(params):
...
+#        Cost after step    65: -1.0000000
+#        Cost after step    70: -1.0000000
+#        Cost after step    75: -1.0000000
+#        Cost after step    80: -1.0000000
+#        Cost after step    85: -1.0000000
+#        Cost after step    90: -1.0000000
+#        Cost after step    95: -1.0000000
+#        Cost after step   100: -1.0000000
+#        Optimized rotation angles: [1.57079633 0.01      ]
+#
```
Comment (trbromley) [inline demonstrations/plugins_hybrid.py:287]: Same issue here

File: demonstrations/plugins_hybrid.py [hunk 4]
```diff
@@ -24,6 +24,9 @@
 Be sure to read through the introductory :ref:`qubit rotation <qubit_rotation>` and
 :ref:`Gaussian transformation <gaussian_transformation>` tutorials before attempting this tutorial.
 
+.. warning::
+    The ``pennylane-sf`` plugin is pinned to ``pennylane<=v0.29``.
```
Comment (trbromley) [inline demonstrations/plugins_hybrid.py:28]:
```suggestion
    This demo is only compatible with PennyLane version `0.29` or below.
```

File: demonstrations/qonn.py [hunk 5]
```diff
@@ -14,6 +14,9 @@
 
 *Author: Theodor Isacsson — Posted: 05 August 2020. Last updated: 08 March 2022.*
 
+.. warning::
+    The ``pennylane-sf`` plugin is pinned to ``pennylane<=v0.29``.
```
Comment (trbromley) [inline demonstrations/qonn.py:18]:
```suggestion
    This demo is only compatible with PennyLane version `0.29` or below.
```

**Review by trbromley — COMMENTED** [review]
Thanks @albi3ro!

File: demonstrations/gbs.py [hunk 6]
```diff
@@ -343,6 +434,16 @@ def gbs_circuit():
...
+# .. rst-class:: sphx-glr-script-out
+#
+#  Out:
+#
+#  .. code-block:: none
+#
+#       0.029573843083205383
+#       0.02957384308320549
+#
 ######################################################################
```
Comment (josh146) [inline demonstrations/gbs.py:447]:
```suggestion
#
```

**Review by mudit2812 — APPROVED** [review]

**Pull request merged**